lib/active_graph/node/query/query_proxy_link.rb

```diff
@@ -111,7 +111,7 @@ def uniq_param_generator_lambda
                 proxy_params.each_with_object({}) do |(param_name, param_val), new_params|
                   new_params_key = "#{prefix}#{param_name}".to_sym
                   new_params[new_params_key] = param_val
-                  proxy_cypher.gsub!(/(?<![`'"])\$#{param_name}(?![^\s)])(?![`'"])/, "$#{new_params_key}")
+                  proxy_cypher.gsub!(/\$#{param_name}(?=(?:[^`'"]|'(?:\\.|[^'])*'|"(?:\\.|[^"])*"|`(?:\\.|[^`])*`)*[^'"`]*\z)/, "$#{new_params_key}")
```

Comment:
Excellent! Just the `?:` are unnecessary as in lookahead no backreferences are created anyway.

I would've probably used lookbehind as I'm thinking about state machine. When I get to the $var I would like to know in what state I am, quoted or unquoted. But that's just a taste. Both alternatives rely on the assumption that the entire string is syntactically correct.

Reply:
Already in lookahead so group won't capture it 👍